### PR TITLE
Support Xcode 12.0.0

### PR DESCRIPTION
**Pull request by mokagio**
Adds support for Xcode 12.0. Everything worked out of the box. Almost too easy, I'm worried.

### Test

Checkout this branch, run it on the latest Xcode 12 beta (beta 6, currently). Also run the unit tests.

### Review

Only one developer is required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

### CODE REVIEW

**Comment by jleandroperez** [comment]
@mokagio sir!! I may have bumped up CircleCI to use Xcode 12 already (recently!!). 

Do we need to change anything here, or it's good to go? (there are a couple extra runtime hardening flags toggled!)

File: Simplenote.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        "state": {
+          "branch": null,
+          "revision": "c594338829e8fb8954fad8743508b135da17b132",
+          "version": "1.0.0"
+        }
+      }
+    ]
+  },
+  "version": 1
+}
```
Comment (mokagio) [inline Simplenote.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: This is not necessarily related to Xcode 12, but I think we should track it. It's the equivalent of the `Podfile.lock`.

**Comment by mokagio** [comment]
Thanks for following up on this @jleandroperez. I rebased on top of your changes. I think we should still merge this PR, even if it's just to get rid of the "Update project to recommended..." warining.

File: Simplenote.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (jleandroperez) [inline Simplenote.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: It's a bit odd for few reasons!!

- Not present in `develop`
- SimplenoteFoundation is pinned to 1.1.0 actually
- And there's a third dependency called SimplenoteInterlinks

Should we just delete this file? (maybe it's a stale thing generated, at some point, by Xcode 11?)

**Review by jleandroperez — APPROVED** [review]
Added one question for you sir!! other than that, feel free to :shipit: !!

Thank you Gio!!

File: Simplenote.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (mokagio) [reply Simplenote.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: @jleandroperez interesting... I thought it was an Xcode 12 thing but I clearly got confused with [the `Package.resolved` from `Simplenote.xcworkspace`](https://github.com/Automattic/simplenote-macos/blob/51dfca83a4d1d1d8c0e7ceee6d9b0e069f0a91e3/Simplenote.xcworkspace/xcshareddata/swiftpm/Package.resolved) which we already track. Having two wouldn't make much sense...

I deleted the file and reopened the workspace, waiting for the Swift packages to resolve. The file was not created. Seems safe to remove – which is what I did with an interactive rebase.

Thanks!

**Pull request merged**